calico.go

```diff
@@ -44,80 +44,91 @@ func init() {
+	// Allow the nodename to be overridden by the network config
+	nodename = utils.DetermineNodename(conf)
 
-	workload, orchestrator, err := GetIdentifiers(args)
+	// Extract WEP identifiers such as pod name, pod namespace (for k8s), containerID, workload name.
+	epIDs, err := utils.GetIdentifiers(args, nodename)
 	if err != nil {
 		return err
 	}
 
-	logger := CreateContextLogger(workload)
-
-	// Allow the nodename to be overridden by the network config
-	updateNodename(conf, logger)
-
-	logger.WithFields(log.Fields{
-		"Orchestrator": orchestrator,
-		"Node":         nodename,
-		"Workload":     workload,
-		"ContainerID":  args.ContainerID,
-	}).Info("Extracted identifiers")
+	logger := &log.Entry{}
+	if epIDs.Orchestrator == "k8s" {
+		logger = log.WithFields(log.Fields{
+			"WorkloadEndpoint": epIDs.Workload,
+			"ContainerID":      epIDs.ContainerID,
+			"Pod":              epIDs.Pod,
+			"Namespace":        epIDs.Namespace,
+		})
+	} else {
+		logger = log.WithFields(log.Fields{
+			"WorkloadEndpoint": epIDs.Workload,
+			"ContainerID":      epIDs.ContainerID,
+		})
+	}
+	logger.Info("Extracted identifiers")
+	logger.WithField("NetConfg", conf).Info("Loaded CNI NetConf")
 
-	logger.WithFields(log.Fields{"NetConfg": conf}).Info("Loaded CNI NetConf")
-	calicoClient, err := CreateClient(conf)
+	calicoClient, err := utils.CreateClient(conf)
 	if err != nil {
 		return err
 	}
 
-	// Always check if there's an existing endpoint.
-	endpoints, err := calicoClient.WorkloadEndpoints().List(api.WorkloadEndpointMetadata{
-		Node:         nodename,
-		Orchestrator: orchestrator,
-		Workload:     workload})
+	ctx := context.Background()
+
+	// Check if there's an existing endpoint.
+	endpoints, err := calicoClient.WorkloadEndpoints().List(ctx, options.ListOptions{Name: epIDs.Workload, Namespace: epIDs.Namespace, Prefix: true})
 	if err != nil {
 		return err
 	}
 
-	logger.Debugf("Retrieved endpoints: %v", endpoints)
-
 	var endpoint *api.WorkloadEndpoint
-	if len(endpoints.Items) == 1 {
-		endpoint = &endpoints.Items[0]
+
+	if len(endpoints.Items) >= 1 {
+		logger.Debugf("List of WorkloadEndpoints %v", endpoints.Items)
+		for _, ep := range endpoints.Items {
+			match, err := epIDs.WorkloadEndpointIdentifiers.NameMatches(ep.Name)
+			if err != nil {
+				// we should never hit this error, because if this should have been
+				// caught by CalculateWorkloadEndpointName.
+				return fmt.Errorf("invalid WorkloadEndpoint identifiers: %v", epIDs.WorkloadEndpointIdentifiers)
+			}
+
+			if match {
+				logger.Debugf("Found a match for WorkloadEndpoint: %v", ep)
+				endpoint = &ep
+				break
+			}
+		}
+
+		if endpoint == nil {
+			return fmt.Errorf("WorklodaEndpoint name does not match the provided identifiers")
+		}
 	}
 
-	fmt.Fprintf(os.Stderr, "Calico CNI checking for existing endpoint: %v\n", endpoint)
+	logger.Debugf("Retrieved endpoints: %v", endpoint)
+	logger.Infof("Calico CNI checking for existing endpoint: %v", endpoint)
```

Comment:
I don't get this log - i thought we'd just checked for existing endpoint?

Reply:
Yea the wording of this log has always been weird.
It should say something like "Calico CNI found existing endpoint: %v"